qualtran/bloqs/mcmt/and_bloq_test.py

```diff
@@ -186,6 +186,8 @@ def signature(self) -> 'Signature':
     def build_composite_bloq(
         self, bb: 'BloqBuilder', q0: 'SoquetT', q1: 'SoquetT'
     ) -> Dict[str, 'SoquetT']:
+        assert isinstance(q0, Soquet)
+        assert isinstance(q1, Soquet)
```

Comment:
is this sort of thing going to be generally necessary? presumably one could also change the method signature to assign these the types `Soquet` instead of `SoquetT`, right?

Reply:
Yes, a lot of these are since SoquetT can be a numpy array of Soquet.  Changing the type of the return signature would also fix this.  

I'm not sure I really have the bandwidth to figure whether this is possible on a case-by-case basis with 1400+ mypy errors to sort through and will generally opt for the simple solution unless reviewers point out cases where we want to change it.

Reply:
What do you mean by changing the return type signature? It's currently `Dict[str, 'SoquetT']` which means these can be single soquets or arrays thereof, no?

Reply:
I mean your original suggestion "presumably one could also change the method signature to assign these the types Soquet instead of SoquetT"